docs/delete-product-or-product-variant-metafields-in-bulk/script.liquid

```diff
@@ -0,0 +1,221 @@
+                    id
+                    description
+                    namespace
+                    key
+                    value
+                    valueType
+                  }
+                {% else %}
+                  metafields
+                    {% if options.metafield_namespace != blank %}
+                      (
+                      namespace: {{ options.metafield_namespace | json }}
+                      ) 
+                    {% endif %}
+                    {
+                    edges {
+                      node {
+                        __typename
+                        id
+                        description
+                        namespace
+                        key
+                        value
+                        valueType
+                      }
+                    }
+                  }
+                {% endif %}
+              }
+            }
+          }
+        }
+    {% endcapture %}
+  {% comment %} Delete product variant metafields query {% endcomment %} 
+  {% elsif options.delete_product_variant_metafields__boolean %}
+     {% capture bulk_operation_query %}
+        query {
+          productVariants {
+            edges {
+              node {
+                __typename
+                id
+                {% if options.metafield_key != blank %}
+                  metafield(
+                    namespace: {{ options.metafield_namespace | json }}
+                    key: {{ options.metafield_key | json }}
+                  ) {
+                    __typename
+                    id
+                    description
+                    namespace
+                    key
+                    value
+                    valueType
+                  }
+                {% else %}
+                 metafields
+                    {% if options.metafield_namespace != blank %}
+                      (
+                      namespace: {{ options.metafield_namespace | json }}
+                      ) 
+                    {% endif %}
+                    {
+                    edges {
+                      node {
+                        __typename
+                        id
+                        description
+                        namespace
+                        key
+                        value
+                        valueType
+                      }
+                    }
+                  }
+                {% endif %}
+              }
+            }
+          }
+        }
+    {% endcapture %}       
+  {% endif %}
+
+  {% action "shopify" %}
+    mutation {
+      bulkOperationRunQuery(
+        query: {{ bulk_operation_query | json }}
+      ) {
+        bulkOperation {
+          id
+          status
+        }
+        userErrors {
+          field
+          message
+        }
+      }
+    }
+  {% endaction %}
+
+{% elsif event.topic == "mechanic/shopify/bulk_operation" %}
+  {% if event.preview %}
+    {% if options.delete_product_metafields__boolean %}
```

Comment:
Not gonna insist on this, but I do suggest doing a similar deduplication thing _here_, swapping between `"Product"` and `"ProductVariant"` in your stub data. Easier to guarantee that everything else operates otherwise the same. :)

Reply:
As discussed, I decided to leave this as is for code readability.